src/Symfony/Component/Routing/RoutePresenceCheckableInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing;
+
+/**
+ * Interface for checking a route presence.
+ *
+ * @author Oleg Voronkovich <oleg-voronkovich@yandex.ru>
+ */
+interface RoutePresenceCheckableInterface
```

Comment:
Can we add this method instead in Symfony 4.0 and not create this `RoutePresenceCheckableInterface`? It looks overkill to me.

Reply:
any hint @javiereguiluz ?

Reply:
I don't know about the internal details, but I'd like to call this method on the `router` object:

```php
// old Symfony
public function indexAction()
{
    if ($this->get('router')->hasRoute('special_route')) {
        // ...
    }
}

// new Symfony
public function indexAction(RouterInterface $router)
{
    if ($router->hasRoute('special_route')) {
        // ...
    }
}
```

Reply:
@javiereguiluz, the `router` implements the `RoutePresenceCheckableInterface` interface, so your code should work fine.

Reply:
Yes. I was suggesting to add it in `UrlGeneratorInterface` (only in Sf 4.0) to avoid creating micro-interfaces.